source/reduce/remove_unreferenced_instruction_reduction_pass.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "remove_unreferenced_instruction_reduction_pass.h"
+#include "remove_instruction_reduction_opportunity.h"
+#include "source/opt/instruction.h"
+
+namespace spvtools {
+namespace reduce {
+
+using namespace opt;
+
+std::vector<std::unique_ptr<ReductionOpportunity>>
+RemoveUnreferencedInstructionReductionPass::GetAvailableOpportunities(
+    opt::IRContext* context) const {
+  std::vector<std::unique_ptr<ReductionOpportunity>> result;
+
+  for (auto& function : *context->module()) {
+    for (auto& block : function) {
+      for (auto& inst : block) {
+        if (context->get_def_use_mgr()->NumUses(&inst) == 0) {
+          switch (inst.opcode()) {
+            case SpvOpBranch:
+            case SpvOpBranchConditional:
+            case SpvOpLoopMerge:
+            case SpvOpSelectionMerge:
+            case SpvOpReturn:
+            case SpvOpSwitch:
+              // TODO: this should ultimately capture all opcodes that relate to
```

Comment:
Missing OpKill, OpReturnValue and OpUnreachable.

Potential alternative code to check:
```
If (!(spvOpcodeIsBlockTerminator(inst.opcode() || inst.opcode() == SpvOpSelectionMerge || inst.opcode() == SpvOpLoopMerge))
```

Reply:
Done, thanks!